openmdao/components/interp_util/interp_akima.py

```diff
@@ -59,20 +59,23 @@ def dv_abs_smooth_complex(x, x_deriv, delta_x):
     # Special case when x is (1, ) and x_deriv is (1, n).
     if len(x_deriv.shape) == 1:
         if x[0] >= delta_x:
-            return x_deriv
+            return x, x_deriv
         elif x[0] <= -delta_x:
-            return -x_deriv
+            return -x, -x_deriv
         else:
-            return 2.0 * x[0] * x_deriv / (2.0 * delta_x)
+            return x[0]**2 / (2.0 * delta_x) + delta_x / 2.0, 2.0 * x[0] * x_deriv / (2.0 * delta_x)
```

Comment:
looks like in the expression after the comma and below in the expression for y_deriv you could save a couple of multiplies because you're multiplying and dividing by 2.0.

Reply:
Fixed